File: Day28/DSA/Find_Lucky_Integer_in_an_Array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int,int> mp;

        for(auto i : arr)
        {
            mp[i]++;
        }

        int lucky = -1;
        for(auto i : mp)
        {
            if(i.first == i.second)
            lucky = max(i.first,lucky);
        }

        return lucky;
    }
};

File: Day28/DSA/Find_Lucky_Integer_in_an_Array.java
Question:-  https://leetcode.com/problems/find-lucky-integer-in-an-array/description/

Code:-  

class Solution {
    public int findLucky(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i : arr) {
            int count = map.getOrDefault(i, 0);
            map.put(i, count + 1);
        }

        int lucky = -1;

        for (Map.Entry<Integer, Integer> ele : map.entrySet()) {
            if (ele.getKey() == ele.getValue()) {
                lucky = Math.max(ele.getKey(), lucky);
            }
        }
        
        return lucky;
    }
}